ISISScientificComputing/autoreduce
# Create Interpreter for plotting meta language

### Summary of work
- This PR creates an interpreter to read plot type files as a _plotly Layout dictionary_.
- Due to design decisions made by the team during the development of this interpreter - its functionality is very minimal.
- It simply loads a plot type file from a given location (.yaml expected by .txt works too) and interprets the data as a python dict.
- If the highest level key of the dict is 'figure', as it is currently in the `example.yaml` file (see #474), then this level is removed and only its value is returned. Otherwise, the whole dict is returned.

![image](https://user-images.githubusercontent.com/31534888/76199674-348a3080-61e8-11ea-8271-54c69fd5b92e.png)

### How to test your work
- Run the accompanying `test_interpreter.py` file and ensure all tests pass
- Run the interpreter with a variety of plotting meta language files to the interpreter and get the desired result. This should include but not be limited to the example file.

### Additional comments
- Since the python `yaml` library loads yaml data as a python dictionary natively, no post-load interpretation is required.

Fixes #475 

**Before merging ensure the release notes have been updated**
